feat(ProviderSettings): enhance provider data management on mount

DeJeune opened the pull request:
- Implemented a cleanup process for provider data to remove duplicates and ensure all system providers are included.
- Added logic to identify and eliminate duplicate providers based on their IDs.
- Integrated missing system providers into the list, ensuring comprehensive provider management upon component mount.

<!-- Template from https://github.com/kubevirt/kubevirt/blob/main/.github/PULL_REQUEST_TEMPLATE.md?-->
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as draft: https://github.com/CherryHQ/cherry-studio/blob/main/CONTRIBUTING.md
-->

### What this PR does

Before this PR:

加载providersettings页面时过滤掉重复的provider，并且补齐缺失的provider

After this PR:

<!-- (optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: -->

Fixes #9574

DeJeune committed:
feat(ProviderSettings): enhance provider data management on mount

- Implemented a cleanup process for provider data to remove duplicates and ensure all system providers are included.
- Added logic to identify and eliminate duplicate providers based on their IDs.
- Integrated missing system providers into the list, ensuring comprehensive provider management upon component mount.

GeorgeDong32 commented on src/renderer/src/pages/settings/ProviderSettings/ProviderSetting.tsx:
```diff
@@ -225,6 +225,52 @@ const ProviderSetting: FC<Props> = ({ providerId }) => {
+      // Find duplicates (same id appears multiple times)
+      const seenIds = new Set<string>()
+      const duplicateIds = new Set<string>()
+      currentProviders.forEach((p) => {
```
这里要不要判一下APIKey是否为空来确定保留哪一个，会不会第一个是错误产生的

DeJeune committed:
refactor(ProviderSettings): streamline provider cleanup logic

- Removed inline cleanup function and utilized a dedicated utility to manage provider data.
- Enhanced the cleanup process to return both cleaned providers and a change flag for better state management.
- Simplified the useEffect hook for improved readability and maintainability.

DeJeune commented on src/renderer/src/pages/settings/ProviderSettings/ProviderSetting.tsx:
```diff
@@ -225,6 +226,16 @@ const ProviderSetting: FC<Props> = ({ providerId }) => {
     )
   }
 
+  // Clean up provider data on component mount - remove duplicates and add missing system providers
+  useEffect(() => {
+    const { cleanedProviders, hasChanges } = cleanupProviders(allProviders)
+
+    if (hasChanges) {
+      updateProviders(cleanedProviders)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []) // Empty dependency array to run only on mount
+
```
意思是手动修复？现在没有ProviderList组件

alephpiece replied:
ProviderSetting 的父组件就是 ProviderList
src\renderer\src\pages\settings\ProviderSettings\index.tsx

alephpiece replied:
打错了，是 ProvidersList

alephpiece commented on src/renderer/src/utils/provider.ts:
```diff
@@ -0,0 +1,87 @@
+    return true
+  }
+
+  return false
```
有没有可能重复的 provider 都有 api key，这时只保留第一个吗？
是不是应该只删除能删除的，不确定的就弹出提示说明一下清空 api key 之后才会删除

alephpiece commented:
靠 api key 和 enabled 来区分，似乎不能处理 vertexai 重复的情况？比如 api host 不同的情况

DeJeune committed:
feat(ProviderSettings): implement provider cleanup with conflict resolution

- Added a new cleanup process for provider data that identifies and resolves duplicates, allowing user intervention for conflicts.
- Introduced a modal for users to select which provider configurations to keep when duplicates are detected.
- Updated the cleanup function to return both cleaned providers and any conflicts that require user attention.
- Enhanced the UI to include a cleanup button and integrated the conflict resolution popup for better user experience.

alephpiece commented:
<img width="321" height="99" alt="image" src="https://github.com/user-attachments/assets/34894422-25d3-4f07-8fcb-7d1e9af5ee4d" />

这功能感觉不是很常用，是不是考虑收到别的地方去？比如 provider list 右键菜单，或者数据设置界面

DeJeune commented:
> <img alt="image" width="321" height="99" src="https://private-user-images.githubusercontent.com/22237751/482625560-34894422-25d3-4f07-8fcb-7d1e9af5ee4d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYyOTM4NTUsIm5iZiI6MTc1NjI5MzU1NSwicGF0aCI6Ii8yMjIzNzc1MS80ODI2MjU1NjAtMzQ4OTQ0MjItMjVkMy00ZjA3LThmY2ItN2QxZTlhZjVlZTRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI3VDExMTkxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0NDU3MDUzOWU2ZDQ3NTczYzQ0OWQyNWIwOTY2MmU3N2I0MDhhZGYwNGI4NmExYzgxY2JjMDg5NzRhY2IzNWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6xi3JNOq7JjOfxLw04t0C0_DdYznSuhwFE7rOHchz1U">
> 这功能感觉不是很常用，是不是考虑收到别的地方去？比如 provider list 右键菜单，或者数据设置界面

没有冲突隐藏起来如何

EurFelux commented:
> > <img alt="image" width="321" height="99" src="https://private-user-images.githubusercontent.com/22237751/482625560-34894422-25d3-4f07-8fcb-7d1e9af5ee4d.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTYyOTM4NTUsIm5iZiI6MTc1NjI5MzU1NSwicGF0aCI6Ii8yMjIzNzc1MS80ODI2MjU1NjAtMzQ4OTQ0MjItMjVkMy00ZjA3LThmY2ItN2QxZTlhZjVlZTRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA4MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwODI3VDExMTkxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0NDU3MDUzOWU2ZDQ3NTczYzQ0OWQyNWIwOTY2MmU3N2I0MDhhZGYwNGI4NmExYzgxY2JjMDg5NzRhY2IzNWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6xi3JNOq7JjOfxLw04t0C0_DdYznSuhwFE7rOHchz1U">
> > 这功能感觉不是很常用，是不是考虑收到别的地方去？比如 provider list 右键菜单，或者数据设置界面
> 
> 没有冲突隐藏起来如何

有冲突的时候可以搞显眼一点，没冲突就隐藏
